ovs/datapath_test.go

```diff
@@ -0,0 +1,644 @@
+	z2["limit"] = 1000000
+	zones := make([]Zone, 0)
+	zones = append(zones, z1)
+	zones = append(zones, z2)
+	out := &ConnTrackOutput{
+		defaultLimit: z0,
+		zones:        zones,
+	}
+
+	var tests = []struct {
+		desc     string
+		dp       *DataPathService
+		dpName   string
+		zones    []uint64
+		want     *ConnTrackOutput
+		err      string
+		testCase uint8
+	}{
+		{
+			desc: "Test invalid ovs-dpctl version ",
+			dp: &DataPathService{
+				CLI: &MockOvsCLI{
+					Version: 9,
+				},
+			},
+			dpName:   "ovs-system",
+			zones:    []uint64{2, 3},
+			want:     nil,
+			err:      "ovs-dpctl: unknown command 'ct-get-limits'; use --help for help",
+			testCase: 9,
+		},
+		{
+			desc: "Test valid ovs-dpctl version ",
+			dp: &DataPathService{
+				CLI: &MockOvsCLI{
+					Version: 10,
+				},
+			},
+			dpName:   "ovs-system",
+			zones:    []uint64{2, 3},
+			want:     out,
+			err:      "",
+			testCase: 10,
+		},
+		{
+			desc: "Test valid ovs-dpctl ct-get-limits",
+			dp: &DataPathService{
+				CLI: &MockOvsCLI{
+					Version: 10,
+				},
+			},
+			dpName:   "",
+			zones:    []uint64{},
+			want:     out,
+			err:      "datapath name argument is mandatory",
+			testCase: 11,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			results, err := tt.dp.GetCTLimits(tt.dpName, tt.zones)
+			switch tt.testCase {
+			case 9:
+				if err.Error() != tt.err {
+					t.Errorf("getting an error while trying to get conntrack limits %q", err.Error())
+				}
+			case 10:
+				if tt.want.defaultLimit["default"] != results.defaultLimit["default"] {
+					t.Errorf("mismatched values, want %q  but got %q", tt.want.defaultLimit["default"], results.defaultLimit["default"])
+				}
+				for i, z := range results.zones {
+					if tt.want.zones[i]["zone"] != z["zone"] {
+						t.Errorf("mismatched values, want %q  but got %q", tt.want.zones[i]["zone"], z["zone"])
+					}
+				}
+			case 11:
+				if err != nil && err.Error() != tt.err {
+					t.Errorf(err.Error())
+				}
+			default:
+				t.Log("pass")
+			}
+		})
+	}
+}
+
+func TestGetCTLimitsWithBinary(t *testing.T) {
+	if testing.Short() {
+		// if you want to run tests in a context where
+		// you don't get ovs-dpctl installed, you should run
+		// go test -short to skip it
+		t.Skip("skipping test in short mode.")
+	}
+
+	z0 := make(Zone)
+	z0["default"] = 200
+	z1 := make(Zone)
+	z1["zone"] = 2
+	z1["count"] = 0
+	z1["limit"] = 200
+	z2 := make(Zone)
+	z2["zone"] = 3
+	z2["count"] = 0
+	z2["limit"] = 200
+	zones1 := make([]Zone, 0)
+	zones1 = append(zones1, z1)
+	zones1 = append(zones1, z2)
+	out := &ConnTrackOutput{
+		defaultLimit: z0,
+		zones:        zones1,
+	}
+	zones2 := make([]Zone, 0)
+	zones2 = append(zones2, z2)
+	outTest2 := &ConnTrackOutput{
+		defaultLimit: z0,
+		zones:        zones2,
+	}
+
+	var tests = []struct {
+		desc     string
+		dp       *DataPathService
+		dpName   string
+		zones    []uint64
+		want     *ConnTrackOutput
+		err      string
+		testCase uint8
```

Comment:
what is the purpose of this variable ? (testCase)

Reply:
It is used to switch inside the `tt`

Reply:
I want to test multiple conditions valid test case and errors are well supported.
`10, 11` are just arbitrary values. I could put `0, 1` for clarity.